egregoria/src/vehicles/systems.rs

```diff
@@ -187,6 +202,14 @@ fn state_update(
                         to_derivative: dir * 2.0,
                     };
 
+                    // Create some points along the spline and repack the itin with the new points.
+                    let mut new_points: Vec<Vec2> = s.points(8).collect();
+                    new_points.append(&mut itin.local_path.clone());
+                    let itin = Itinerary {
```

Comment:
instead of making a new itinerary, I'd prefer to mutate itin directly, maybe add a method like prepend_points ? This also solves the issue of not making itinerary's attributes public